Mage.Sets/src/mage/cards/g/GaviNestWarden.java

```diff
@@ -0,0 +1,141 @@
+package mage.cards.g;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.DrawSecondCardTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.effects.common.cost.CostModificationEffectImpl;
+import mage.abilities.keyword.CyclingAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterAbility;
+import mage.filter.predicate.Predicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.token.DinosaurCatToken;
+import mage.watchers.Watcher;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ *
+ * @author htrajan
+ */
+public final class GaviNestWarden extends CardImpl {
+
+    private static final FilterAbility filter = new FilterAbility("first card cycled each turn");
+
+    static {
+        filter.add(new FirstCycledCardPredicate());
+    }
+
+    public GaviNestWarden(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{U}{R}{W}");
+        
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SHAMAN);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(5);
+
+        // You may pay {0} rather than pay the cycling cost of the first card you cycle each turn.
+        Effect effect = new CyclingZeroCostEffect(filter);
+        effect.setText("You may pay {0} rather than pay the cycling cost of the first card you cycle each turn.");
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, effect), new GaviNestWardenWatcher());
```

Comment:
You can set the text for the effect in the class definition instead of up here, also you don't need to specify the zone in the constructor for `SimpleStaticAbility` since it's the default.

Reply:
fixed!